mesonbuild/modules/gnome.py

```diff
@@ -1350,8 +1350,10 @@ def abs_filenames(files: T.Iterable['FileOrString']) -> T.Iterator[str]:
             t_args.append(f'--{program_name}={path}')
         if namespace:
             t_args.append('--namespace=' + namespace)
-        if state.environment.need_exe_wrapper() and not isinstance(state.environment.get_exe_wrapper(), EmptyExternalProgram):
-            t_args.append('--run=' + ' '.join(state.environment.get_exe_wrapper().get_command()))
+        # if not need_exe_wrapper, we get an EmptyExternalProgram. If none provided, we get NoneType
+        exe_wrapper = state.environment.get_exe_wrapper()
+        if not isinstance(exe_wrapper, (NoneType, EmptyExternalProgram)):
```

Comment:
Couldn't this be `if exe_wrapper and exe_wrapper.found():`

Reply:
No it cannot, because an EmptyExternalProgram will exist (not NoneType) ***and*** is found, it just evaluates to `self.command = []`.

Reply:
What the heck is an `EmptyExternalProgram` and why do we have it? *sigh*